openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,839 @@
       along with this program.  If not, see <https://www.gnu.org/licenses/agpl.html>.
 */
-
-var utfGrid = {};
-var layerControl;
-var selectedPDef;
-var previousCatData = [];
-
-var TILESTREAM_URL = TS_URL + '/v2/';
-var TILESTREAM_API_URL = TS_URL + '/api/v1/Tileset/';
-
-// Keep track of the layer names
-var layers;
-
-// Make a list of categorys
-var categoryList = [];
-var layersByCat = {};
-var layerNames = {};
-var layerGrids = [];
-var projectDefinition = {};
-
-// Indicators for the PCP charts
-var primaryIndicator = {};
-var categoryIndicator = {};
-var sessionPrimaryIndicator = {};
-var sessionCategoryIndicator = {};
-var weightedPrimaryIndicator = {};
-var catIndicator = {};
-var weightedCatIndicator = {};
-var scaledCatIndicator = {};
-var aalIndicator = {};
-var scaledAalIndicator = {};
-var weightedAalIndicator = {};
-var sviIndicator = {};
-var scaledSviIndicator = {};
-var weightedSviIndicator = {};
-var iriIndicator = {};
-
-// vars used to set the weights of the project definition json
-var pdData;
-
-// Keep track of project definition elements whos weights have been changes
-var tempWeight = {};
-var tempCatWeight = {};
-var pdTempPrimaryIndicator = [];
-var pdTempPrimaryIndicatorLevels = [];
-var pdTempPILevel = {};
-var pdTempCategoryIndicator = [];
-var parentChildKey = {};
-var sviParentChildKey = {};
-var pdTempCatWeight = {};
-var tempParentChildKey = [];
-var catIndicator = {};
-var tempCatSearchElements = [];
-var tempSviSearchElements = [];
-var tempSviWeight = '';
-var tempAalWeight = '';
-var tempAalValue = '';
-var tempIriWeight = '';
-var municipality = [];
-var outlierBreakPoint = 0.75; // used for the primary indicator outlier
-
-// Keep track of the utfGrid that has been selected last
-var selectedGrid;
+var layerAttributes;
+var sessionProjectDef = [];
+var selectedRegion;
+
+// sessionProjectDef is the project definition as is was when uploaded from the QGIS tool.
+// While projectDef includes modified weights and is no longer the version that was uploaded from the QGIS tool
+var sessionProjectDefStr;
+var projectLayerAttributes;
+var regions = [];
 var baseMapUrl = new L.TileLayer('http://otile1.mqcdn.com/tiles/1.0.0/map/{z}/{x}/{y}.png');
-
 var app = new OQLeaflet.OQLeafletApp(baseMapUrl);
-
-var startApp = function() {
-    map = new L.Map('map', {
-        minZoom: 2,
-        scrollWheelZoom: false,
-        attributionControl: false,
-        maxBounds: new L.LatLngBounds(new L.LatLng(-90, -180), new L.LatLng(90, 180)),
-    });
-    map.setView(new L.LatLng(10, -10), 2).addLayer(baseMapUrl);
-
-    layers = {};
-
-    layerControl = L.control.layers(app.baseLayers);
-    map.panTo(new L.LatLng(39.399, -8.22));
-    map.setZoom(6);
-
-    // Duplicate layer warnning message
-    function showDuplicateMsg() {
-        $('#warning-duplicate').dialog('open');
+var indicatorChildrenKey = [];
+
+function createIndexSimple(la, riskIndicators) {
```

Comment:
I've just noticed you have both `createIndexSimple` and `createIndex`. Can you write somewhere what's the difference and in which cases they are used?